### PR TITLE
ISPN-11851 FD_SOCK should use the same bind address as TCP

### PR DESCRIPTION
**Pull request by danberindei**
https://issues.redhat.com/browse/ISPN-11851
https://issues.redhat.com/browse/ISPN-12149

Also deprecate system properties jgroups.tcp.address
and jgroups.udp.address

### CODE REVIEW

File: core/src/test/resources/configs/unified/11.0.xml [hunk 1]
```diff
@@ -10,12 +10,12 @@
       <stack-file name="tcp-test" path="stacks/tcp_mping/tcp1.xml"/>
       <!-- Inline definition -->
       <!-- Inline definition -->
-      <stack name="mping">
+      <stack name="mping" extends="tcp">
```
[the remark this answers is not in the document]
Comment (wburns) [reply core/src/test/resources/configs/unified/11.0.xml:13]: Also if we are extending tcp, we should probably only add the stack elements that are actually changing something. Most of the ones below are exactly the same.

File: documentation/src/main/asciidoc/topics/config_examples/config_inline_jgroups.xml [hunk 2]
```diff
@@ -1,8 +1,8 @@
 <infinispan> <1>
   <jgroups> <2>
-    <stack name="prod"> <3>
+    <stack name="prod" extends="tcp"> <3>
```
Comment (wburns) [inline documentation/src/main/asciidoc/topics/config_examples/config_inline_jgroups.xml:3]: Same thing regarding trimming the rest of the protocols that are duplicate between prod and tcp.

File: core/src/test/resources/configs/unified/11.0.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (danberindei) [reply core/src/test/resources/configs/unified/11.0.xml:13]: Yeah, I'll remove the common ones. I tried to expedite things by only removing the bind address stuff, but in general it's a good idea to remove the duplication in the test XMLs even if the cache manager is never started.

File: documentation/src/main/asciidoc/topics/config_examples/config_inline_jgroups.xml (hunk 2 above)
Comment (danberindei) [reply documentation/src/main/asciidoc/topics/config_examples/config_inline_jgroups.xml:3]: I think I also need to modify the asciidoc here, that's why I didn't make more extensive changes.

File: core/src/test/resources/configs/unified/11.0.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (danberindei) [reply core/src/test/resources/configs/unified/11.0.xml:13]: Actually there's another stack extending `tcp` just below, so I'll keep the full definition here and only remove the `bind_addr` attribute.

[the remark this answers is not in the document]
Comment (danberindei) [reply core/src/test/resources/configs/unified/11.0.xml]: WRT `FD_ALL3` and `FD_ALL`, `FD_ALL3` is very new, and that's why I only added it in the test configurations. In the original version with `extends="tcp"`, I should have used `stack.combine="REPLACE" stack.position="FD_ALL"`.

**Comment by danberindei** [comment]
Updated to remove the `bind_addr` attribute from all non-transport protocols like `FD_SOCK` and `MPING`.

**Comment by tristantarrant** [comment]
JsonSerializationTest.testJGroups:346 expected:<9> but was:<10>

**Comment by danberindei** [comment]
Updated to fix `JsonSerializationTest`.

File: core/src/main/resources/default-configs/default-jgroups-azure.xml [hunk 3]
```diff
@@ -5,6 +5,7 @@
 <config xmlns="urn:org:jgroups"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="urn:org:jgroups http://www.jgroups.org/schema/jgroups-4.2.xsd">
+   <!-- jgroups.tcp.address is deprecated and will be removed, see ISPN-11867 -->
    <TCP bind_addr="${jgroups.bind.address,jgroups.tcp.address:SITE_LOCAL}"
         bind_port="${jgroups.bind.port,jgroups.tcp.port:7800}"
```
Comment (diegolovison) [inline core/src/main/resources/default-configs/default-jgroups-azure.xml:10]: How about `jgroups.tcp.port`. Should we remove as well ?

Comment (danberindei) [reply core/src/main/resources/default-configs/default-jgroups-azure.xml:10]: There are 2 problems with `jgroups.tcp.address`:
1. It's reused by other protocols in the stack, which may or may not use TCP sockets
2. It's arbitrary, there's no reason someone would want a different IP address for TCP vs UDP.

`jgroups.tcp.port` doesn't have any of those problems: each protocol using TCP sockets directly must have a different port, so the system property cannot be reused.

**Pull request merged**